pkg/controller/registry/resolver/runtime_constraints/provider.go

```diff
@@ -0,0 +1,80 @@
+package runtime_constraints
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry/resolver/cache"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/pkg/errors"
+)
+
+const (
+	maxRuntimeConstraints = 10
```

Comment:
Reasonable restriction to avoid abusing. Make sure this number is known and noticed upon the completion of this work.